drivers: remove on-stack copying of dma cfg on stm32 drivers

mariopaja opened the pull request:
This change removes the on-stack copying and instead uses a pointer to the configuration to  avoid unnecessary stack usage

mathieuchopstm reviewed (commented):
LGTM except for the remark of @gautierg-st

mariopaja committed:
drivers: remove on-stack copying of dma cfg on stm32 drivers

This change removes the on-stack copying and instead
uses a pointer to the configuration to  avoid unnecessary
stack usage

Signed-off-by: Mario Paja <mariopaja@hotmail.com>

mariopaja committed:
drivers: video: stm32_dcmi: relocate dma stream to driver data

This change moves the dma stream from _config to _data,
allowing direct reference the DMA stream rather than
duplicating it on the stack. Additionally, it aligns the
declaration to the other STM32 drivers.

Signed-off-by: Mario Paja <mariopaja@hotmail.com>

avolmat-st reviewed (changes requested):
Very minor point so I am also ok to approve it as is. Since dma pointer in introduced in the stm32_dma_init function, one of dma_cfg pointer can directly be retrieved via the dma pointer as done in all other places of that function.

JarmouniA reviewed (commented):
This kind of change should be examined carefully, not saying it is wrong.
It's not just a matter of stack-usage, it's also a question of: Do we want the subsequent values assigned to the members of the stack-allocated struct to be setted globally, or just be passed to the functions called locally with the struct as an arg and then be forgotten?
Since the drivers in qst seem to be working fine without those values being setted globally (unless there are undiscovered bugs, or the values are setted elsewhere), I kind of question the need to set them in the global config/data struct.

mariopaja commented:
@gautierg-st I will have a look into it
